apps/wallet/src/ui/app/components/receipt-card/index.tsx

```diff
@@ -81,15 +82,42 @@ function ReceiptCard({ txn, activeAddress }: ReceiptCardProps) {
         return amount ? Math.abs(amount) : null;
     }, [transferAmount]);
 
-    const isSender = activeAddress === getTransactionSender(txn);
     const isStakeTxn =
         moveCallLabel === 'Staked' || moveCallLabel === 'Unstaked';
 
-    const nftObjectLabel = transferAmount?.length
-        ? isSender
-            ? 'Sent'
-            : 'Received'
-        : 'Call';
+    const { owner } = getGasData(txn);
+    const transactionSender = getTransactionSender(txn);
+    const isSender = activeAddress === transactionSender;
+    const isSponsoredTransaction = transactionSender !== owner;
+    const gasTotal = getTotalGasUsed(txn);
+
+    const showGasSummary = isSuccessful && isSender && gasTotal;
+    const showSponsorInfo = !isSuccessful && isSender && isSponsoredTransaction;
+
+    let txnGasSummary: JSX.Element | undefined;
+    if (showGasSummary && isSponsoredTransaction) {
+        txnGasSummary = (
```

Comment:
I wonder if we do something like this and inline `txnGasSummary`
```suggestion
        txnGasSummary = isSponsoredTransaction ? <SponsoredTxnGasSummary sponsor={owner} totalGas={gasTotal} /> : <TxnGasSummary
                 totalGas={gasTotal}
                 transferAmount={totalSuiAmount}
             />
```